File: 인범/2026_04_29/hotter.py
def solution(scoville, K):
    import heapq as hq
    hq.heapify(scoville)
    answer = 0
    
    while scoville:
        if scoville[0] < K and len(scoville)>=2:
            lowest = hq.heappop(scoville)
            lower = hq.heappop(scoville)
            hq.heappush(scoville, lowest + lower*2)
            answer += 1
        elif scoville[0] >= K: return answer
        else : return -1
    

    return answer

File: 인범/2026_04_29/hotter.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for(int i=0; i<scoville.length; i++){
            pq.add(scoville[i]);
        }
        
        while(true){
            if(pq.peek()<K && pq.size()>=2){
                int lowest = pq.poll();
                int lower = pq.poll();
                int comp = lowest + lower*2;
                pq.add(comp);
                answer++;
            }  
            else{
                break;
            }
        }
        if(pq.peek()>=K){return answer;}
        else{return -1;}
    }
}